src/webgl/framebuffer.js

```diff
@@ -269,11 +270,23 @@ export default class Framebuffer extends Resource {
     return this;
   }
 
+  // Reads pixel data either into an Array object (synchronus) or
+  // into a Buffer object (asynchronus) depending on options and browser capability.
+  readPixels(opts) {
+    if (!isWebGL2(this.gl) || !opts.buffer) {
+      if (opts.buffer) {
+        log.warn('readPixels: No WebGL2 to support, falling back to synchronous version');
```

Comment:
Not sure this makes sense. The app expects a `Buffer` right? Are we just sending back something else for the app to figure out?

Reply:
In fallback mode, we create and return `Array` object. Otherwise we return `Buffer` object. App can figure out based on return object type, I think I should add these details to documentation.

Reply:
My preference would be to simply offer a `readPixels` that reads to array, and a `readPixelsToBuffer` that reads to buffer.

Reply:
Actually I started with that approach but decided to overload similar to other places we do. But this might required apps to process output differently, will keep existing `readPixels` as is and add new `readPixelsToBuffer` as suggested.